### DISCUSSION TITLE
Do not use object ids in code like Codeunit.Run()

### DISCUSSION THREAD

**Discussion by StefanMaron** [StefanMaron/BusinessCentral.LinterCop#16] [New Rule]
This goes for 

- Codeunit.Run()
- Report.Run()
- Report.Execute()
- Report.Save...()
- XMLPort.Export/Import/Run()
- Page.Run/RunModal()
- RecRef.Open()

**Comment by fvet** [comment]
Please also include the same for **EventSubscribers**:

![image](https://user-images.githubusercontent.com/25268332/134901240-2b80d24a-e832-4c2c-915c-781813fd6bd2.png)

**Comment by StefanMaron** [comment]
None of these should compile. But they do, and we only get told at runtime that someone goofed.

I get that the base Page object takes generic var Record: Table arguments, so at the base level those can't be validated - but it is hopefully possible for specific Pages, which have a specific SourceTable set, to error out if they are passed a Table that is not of the same type.

Please can you consider doing this? It would catch common thinkos and save much time.

I might've missed some methods taking Record: Table arguments from my example, in which case, please include all such methods in this fix.

![image](https://user-images.githubusercontent.com/24838311/134908014-f005075b-5a73-423d-a3f2-bb8a182f67b0.png)
![image](https://user-images.githubusercontent.com/24838311/134907911-b53e76dd-f4c2-40c6-b436-89da8da2709c.png)

https://github.com/microsoft/AL/issues/6438
https://github.com/microsoft/AL/issues/6525 (closed, but see my follow-up comment)

https://experience.dynamics.com/ideas/idea/?ideaid=df8e9c09-3986-eb11-8ced-0003ff45a07d

Posted by https://github.com/dnpb

**Reply by StefanMaron** [reply]
I tried to get this to work as well but it seems its not that easy.

**Comment by StefanMaron** [comment]
Boom!
![image](https://user-images.githubusercontent.com/24838311/137377175-9edc0c91-c953-4ff8-88f8-0d4b56bbdbee.png)

**Reply by fvet** [reply]
> Boom! ![image](https://user-images.githubusercontent.com/24838311/137377175-9edc0c91-c953-4ff8-88f8-0d4b56bbdbee.png)

Very nice! Are you considering adding EventSubscribers as well?

**Reply by StefanMaron** [reply]
Ah, I forgot that one :D I will look into it!

**Reply by rvanbekkum** [reply]
Nice! 😊
